Source/Core/Common/IniFile.cpp

```diff
@@ -329,7 +341,8 @@ bool IniFile::Save(const std::string& filename)
     if (!section.keys_order.empty() || !section.m_lines.empty())
       out << '[' << section.name << ']' << std::endl;
 
-    if (section.keys_order.empty())
+    // hack to support the concept of the gameconfig.txt inside of the ini
+    if (section.keys_order.empty() || section.name == "Gecko_GameConfig")
```

Comment:
Probably not required as `keys_order` would be empty, if `[Gecko_GameConfig]` has no key-value pairs.

Reply:
See my comment above.